controlplane/kubeadm/internal/controllers/helpers_test.go

```diff
@@ -539,53 +555,132 @@ func TestKubeadmControlPlaneReconciler_generateMachine(t *testing.T) {
-		NodeDrainTimeout:        &metav1.Duration{Duration: 10 * time.Second},
-	}
-	r := &KubeadmControlPlaneReconciler{
-		Client:            fakeClient,
-		managementCluster: &internal.Management{Client: fakeClient},
-		recorder:          record.NewFakeRecorder(32),
-	}
-	g.Expect(r.generateMachine(ctx, kcp, cluster, infraRef, bootstrapRef, nil)).To(Succeed())
 
-	machineList := &clusterv1.MachineList{}
-	g.Expect(fakeClient.List(ctx, machineList, client.InNamespace(cluster.Namespace))).To(Succeed())
-	g.Expect(machineList.Items).To(HaveLen(1))
-	machine := machineList.Items[0]
-	g.Expect(machine.Name).To(HavePrefix(kcp.Name))
-	g.Expect(machine.Namespace).To(Equal(kcp.Namespace))
-	g.Expect(machine.OwnerReferences).To(HaveLen(1))
-	g.Expect(machine.OwnerReferences).To(ContainElement(*metav1.NewControllerRef(kcp, controlplanev1.GroupVersion.WithKind("KubeadmControlPlane"))))
-	g.Expect(machine.Annotations).To(HaveKeyWithValue(controlplanev1.RemediationForAnnotation, "foo"))
-	g.Expect(kcp.Annotations).ToNot(HaveKey(controlplanev1.RemediationInProgressAnnotation))
-	g.Expect(machine.Spec).To(Equal(expectedMachineSpec))
-
-	// Verify that the machineTemplate.ObjectMeta has been propagated to the Machine.
-	for k, v := range kcpMachineTemplateObjectMeta.Labels {
-		g.Expect(machine.Labels[k]).To(Equal(v))
-	}
-	g.Expect(machine.Labels[clusterv1.ClusterNameLabel]).To(Equal(cluster.Name))
-	g.Expect(machine.Labels[clusterv1.MachineControlPlaneLabel]).To(Equal(""))
-	g.Expect(machine.Labels[clusterv1.MachineControlPlaneNameLabel]).To(Equal(kcp.Name))
-
-	for k, v := range kcpMachineTemplateObjectMeta.Annotations {
-		g.Expect(machine.Annotations[k]).To(Equal(v))
-	}
-
-	// Verify that machineTemplate.ObjectMeta in KCP has not been modified.
-	g.Expect(kcp.Spec.MachineTemplate.ObjectMeta.Labels).NotTo(HaveKey(clusterv1.ClusterNameLabel))
-	g.Expect(kcp.Spec.MachineTemplate.ObjectMeta.Labels).NotTo(HaveKey(clusterv1.MachineControlPlaneLabel))
-	g.Expect(kcp.Spec.MachineTemplate.ObjectMeta.Labels).NotTo(HaveKey(clusterv1.MachineControlPlaneNameLabel))
-	g.Expect(kcp.Spec.MachineTemplate.ObjectMeta.Annotations).NotTo(HaveKey(controlplanev1.KubeadmClusterConfigurationAnnotation))
+	t.Run("should return the correct Machine object when creating a new Machine", func(t *testing.T) {
+		g := NewWithT(t)
+
+		failureDomain := pointer.String("fd1")
+		createdMachine, err := (&KubeadmControlPlaneReconciler{}).computeDesiredMachine(
+			kcp, cluster,
+			infraRef, bootstrapRef,
+			failureDomain, nil,
+		)
+		g.Expect(err).To(BeNil())
+
+		expectedMachineSpec := clusterv1.MachineSpec{
+			ClusterName: cluster.Name,
+			Version:     pointer.String(kcp.Spec.Version),
+			Bootstrap: clusterv1.Bootstrap{
+				ConfigRef: bootstrapRef.DeepCopy(),
+			},
+			InfrastructureRef:       *infraRef.DeepCopy(),
+			FailureDomain:           failureDomain,
+			NodeDrainTimeout:        kcp.Spec.MachineTemplate.NodeDrainTimeout.DeepCopy(),
```

Comment:
nit: Let's drop the deepcopy calls if we don't need them (same the next two lines and in other cases)